packages/paste-core/components/toggle/src/index.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import type {ToggleProps} from './types';
+import type {BackgroundColor, TextColor} from '@twilio-paste/style-props';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {RequiredDot} from '@twilio-paste/label';
+import {ToggleKnob} from './ToggleKnob';
+import {ToggleHelpText} from './ToggleHelpText';
+import {ToggleLabelText} from './ToggleLabelText';
+
+const Toggle = React.forwardRef<HTMLDivElement, ToggleProps>(
+  (
+    {element = 'TOGGLE', children, helpText, id, disabled = false, on = false, required = false, onClick, ...props},
+    ref
+  ) => {
+    const [toggleIsOn, setToggleIsOn] = React.useState(on);
+    const [isHovering, setIsHovering] = React.useState(false);
+
+    if (!id) {
+      const toggleId = useUID();
+      id = toggleId;
+    }
+    const helpTextId = useUID();
+    const labelId = useUID();
+
+    const getBackgroundColor = (): BackgroundColor => {
+      if (!disabled) {
+        if (!toggleIsOn) {
+          if (!isHovering) return 'colorBackgroundStronger';
+          else if (isHovering) return 'colorBackgroundStrongest';
+        } else if (toggleIsOn) {
+          if (!isHovering) return 'colorBackgroundPrimaryStronger';
+          else if (isHovering) return 'colorBackgroundPrimary';
+        }
+      } else if (disabled) {
+        return 'colorBackgroundStrong';
+      }
+      return 'colorBackgroundStronger';
+    };
+
+    const getColor = (): TextColor => {
+      if (disabled) return 'colorTextIcon';
+      if (toggleIsOn && !isHovering) return 'colorTextLinkStronger';
+      if (toggleIsOn && isHovering) return 'colorTextLink';
+      else return 'colorTextIcon';
+    };
+
+    const getCursor = (): string => {
+      if (disabled) return 'not-allowed';
+      if (isHovering) return 'pointer';
+      return 'default';
+    };
+
+    const handleClick = (): void => {
+      if (!disabled) setToggleIsOn(!toggleIsOn);
+      if (onClick) onClick();
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent): void => {
```

Comment:
suggestion: let's wrap these functions in a `useCallback()` to prevent child re-renders